File: backend/internal/model/agent.go
package model

// SynthesisTaskTitle is the reserved title for the final answer synthesis task
const SynthesisTaskTitle = "整合答案"

// AgentTask represents a planned subtask
type AgentTask struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	ToolName    string `json:"tool_name,omitempty"`
	Status      string `json:"status"` // pending/running/done/failed
}

// ToolCall represents a tool invocation
type ToolCall struct {
	Name   string         `json:"name"`
	Args   map[string]any `json:"args"`
	Result *ToolResult    `json:"result,omitempty"`
}

// ToolResult represents the result of a tool call
type ToolResult struct {
	Success bool   `json:"success"`
	Data    any    `json:"data"`
	Error   string `json:"error,omitempty"`
}

// AgentEvent extends SSEEvent for agent reasoning steps
// New event types: thought, task_progress, tool_call, tool_result, final_answer
type AgentEvent struct {
	Type    string `json:"type"`
	Content string `json:"content"`
	Data    any    `json:"data,omitempty"`
}

File: frontend/src/types/agent.ts
export interface AgentTask {
  id: string
  title: string
  description: string
  tool_name?: string
  status: 'pending' | 'running' | 'done' | 'failed'
}

export interface ToolCall {
  call_id: string
  name: string
  args: Record<string, unknown>
  result?: ToolResult
}

export interface ToolResult {
  success: boolean
  data: unknown
  error?: string
}

export interface ReasoningStep {
  step_number: number
  thought: string
  action?: ToolCall
  observation?: string
  status: 'thinking' | 'acting' | 'observing' | 'done'
}

export interface AgentEvent {
  type: 'meta' | 'thought' | 'task_progress' | 'tool_call' | 'tool_result' | 'final_answer' | 'error'
  content: string
  data?: unknown
}

export interface AgentState {
  isAgentMode: boolean
  isStreaming: boolean
  steps: ReasoningStep[]
  tasks: AgentTask[]
  toolCalls: ToolCall[]
  finalAnswer: string
  error: string | null
}

export interface AgentAnalyzeResult {
  needs_agent: boolean
  reason: string
}
